UI/EncyclopediaDetailPanel.cpp

```diff
@@ -2428,7 +2428,28 @@ namespace {
         // show image of planet environment at the top of the suitability report
         std::string planet_type = boost::lexical_cast<std::string>(planet->Type());
         boost::algorithm::to_lower(planet_type);
-        detailed_description += "<img src=\"encyclopedia/planet_environments/" + planet_type + ".png\"></img>";
+
+        namespace fs = boost::filesystem;
+        std::vector<std::string> filenames;
+        fs::directory_iterator end_it;
+        for (fs::directory_iterator it(ClientUI::ArtDir() / "encyclopedia" / "planet_environments"); it != end_it; ++it) {
+            try {
+                if (fs::exists(*it) && !fs::is_directory(*it)) {
```

Comment:
`!is_directory(*it)` -> `is_regular_file(it->status())`
(other types than just directory and file, iteration may provide stat as by-product)

The entire iteration should probably be enclosed in try/catch

Reply:
If the entire for loop is enclosed in try-catch, then permission denied errors would result in all files after it being skipped, instead of just the file for which permission was denied

Reply:
Guessing that is already an issue during iteration?
There are various other instances of directory iterations without a try/catch currently, so probably not a huge concern with an additional case (hope to eventually consolidate filesystem operations, but slow going).

Reply:
I don't believe it's a big issue either way, but I'm not too knowledgable here. Like I said this is basically the same code as `ClientUI::PrefixedTexturesAndDist`, so I figure if it's good enough for that, it's good enough here too. Or at least, pending a consolidation of filesystem operations like you're hoping for.